src/Uno.UI.RuntimeTests/Tests/Windows_UI_Xaml_Controls/Given_Flyout.cs

```diff
@@ -762,6 +767,630 @@ void OnClosing(object sender, FlyoutBaseClosingEventArgs args)
+				closing4 True True True True True True True True True True False
+				closing5 True True True True True True True True True True False
+				closed1 False False False False False 
+				closing2 False False False False False  False
+				closing3 False False False False False  False
+				closing4 False False False False False  False
+				closing5 False False False False False  False
+				closed2 False False False False False 
+				closing3 False False False False False  False
+				closing4 False False False False False  False
+				closing5 False False False False False  False
+				closed3 False False False False False 
+				closing4 False False False False False  False
+				closing5 False False False False False  False
+				closed4 False False False False False 
+				closing5 False False False False False  False
+				closed5 False False False False False 
+
+				""";
+
+				Assert.AreEqual(expected.Replace(Environment.NewLine, "\n"), output);
+			}
+			finally
+			{
+				flyout1.Hide();
+				flyout2.Hide();
+				flyout3.Hide();
+				flyout4.Hide();
+				flyout5.Hide();
+			}
+		}
+
+		[TestMethod]
+		[RunsOnUIThread]
+#if __MACOS__
+		[Ignore("Currently fails on macOS, part of #9282 epic")]
+#endif
+		public async Task When_Opening_And_Closing_Nested_Flyouts_Not_Open()
+		{
+			var flyout1 = new Flyout();
+			var flyout2 = new Flyout();
+			try
+			{
+				var button1 = new Button
+				{
+					Content = "button1",
+					Flyout = flyout1
+				};
+
+				var button2 = new Button
+				{
+					Content = "button2",
+					Flyout = flyout2
+				};
+
+				flyout1.Content = button2;
+				flyout2.Content = new TextBox { Text = "text" };
+
+				var output = "";
+
+				flyout1.Closing += (_, args) => output += $"closing1 {flyout1.IsOpen} {flyout2.IsOpen} {GetAllIsOpens(flyout1)} {args.Cancel}\n";
+				flyout2.Closing += (_, args) => output += $"closing2 {flyout1.IsOpen} {flyout2.IsOpen} {GetAllIsOpens(flyout2)} {args.Cancel}\n";
+
+				flyout1.Closed += (_, _) => output += $"closed1 {flyout1.IsOpen} {flyout2.IsOpen} {GetAllIsOpens(flyout1)}\n";
+				flyout2.Closed += (_, _) => output += $"closed2 {flyout1.IsOpen} {flyout2.IsOpen} {GetAllIsOpens(flyout2)}\n";
+
+				TestServices.WindowHelper.WindowContent = button1;
+				await TestServices.WindowHelper.WaitForIdle();
+
+				flyout1.ShowAt(button1);
+				await TestServices.WindowHelper.WaitForIdle();
+
+				// flyout2 is not open, so Closing should NOT be invoked.
+
+				flyout1.Hide();
+				await TestServices.WindowHelper.WaitForIdle();
+
+				var expected =
+				"""
+				closing1 True False True False
+				closed1 False False 
+
+				""";
+
+				Assert.AreEqual(expected.Replace(Environment.NewLine, "\n"), output);
+			}
+			finally
+			{
+				flyout1.Hide();
+				flyout2.Hide();
+			}
+		}
+
+		[TestMethod]
+		[RunsOnUIThread]
+#if __MACOS__
+		[Ignore("Currently fails on macOS, part of #9282 epic")]
+#endif
+		public async Task When_Opening_And_Closing_Nested_Flyouts_Canceled()
```

Comment:
I can't pass this test at all. Everything about nested flyout/popup dismissal seems identical to WinUI except for this.